plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -0,0 +1,385 @@
+
+    public TrinoRestCatalog(
+            RESTSessionCatalog sessionCatalog,
+            CatalogName catalogName,
+            String trinoVersion,
+            boolean useUniqueTableLocation)
+    {
+        this.sessionCatalog = sessionCatalog;
+        this.catalogName = catalogName;
+        this.trinoVersion = trinoVersion;
+        this.useUniqueTableLocation = useUniqueTableLocation;
+    }
+
+    @Override
+    public List<String> listNamespaces(ConnectorSession session)
+    {
+        return sessionCatalog.listNamespaces(convert(session)).stream()
+                .map(Namespace::toString)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public void dropNamespace(ConnectorSession session, String namespace)
+    {
+        sessionCatalog.dropNamespace(convert(session), Namespace.of(namespace));
+    }
+
+    @Override
+    public Map<String, Object> loadNamespaceMetadata(ConnectorSession session, String namespace)
+    {
+        return ImmutableMap.copyOf(sessionCatalog.loadNamespaceMetadata(convert(session), Namespace.of(namespace)));
+    }
+
+    @Override
+    public Optional<TrinoPrincipal> getNamespacePrincipal(ConnectorSession session, String namespace)
+    {
+        return Optional.empty();
+    }
+
+    @Override
+    public void createNamespace(ConnectorSession session, String namespace, Map<String, Object> properties, TrinoPrincipal owner)
+    {
+        checkArgument(!properties.containsKey(LOCATION_PROPERTY), "Database cannot be created with a location set");
+
+        sessionCatalog.createNamespace(
+                convert(session),
+                Namespace.of(namespace),
+                properties.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, v -> v.getValue().toString())));
+    }
+
+    @Override
+    public void setNamespacePrincipal(ConnectorSession session, String namespace, TrinoPrincipal principal)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "setNamespacePrincipal is not supported for Iceberg REST catalog");
+    }
+
+    @Override
+    public void renameNamespace(ConnectorSession session, String source, String target)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming schemas");
+    }
+
+    @Override
+    public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> namespace)
+    {
+        SessionContext sessionContext = convert(session);
+        List<Namespace> namespaces = namespace.map(List::of)
+                .orElseGet(() -> listNamespaces(session))
+                .stream()
+                .map(Namespace::of)
+                .filter(ns -> sessionCatalog.namespaceExists(sessionContext, ns))
+                .toList();
+
+        ImmutableList.Builder<SchemaTableName> tables = ImmutableList.builder();
+        for (Namespace ns : namespaces) {
+            tables.addAll(
+                    sessionCatalog.listTables(sessionContext, ns)
```

Comment:
Add a try/catch where the catch is a NotFoundException which just ignores the error and continues

This is for the case where the namespace is dropped between the listNamespaces call above and here